From 8e344c7495f38559c4ab23692f86fc7a78ee551a Mon Sep 17 00:00:00 2001
From: Sebastian Bauer <mail@sebastianbauer.info>
Date: Fri, 22 Jan 2016 20:04:50 +0100
Subject: [PATCH 08/20] Added libstc++ support for AmigaOS.

---
 .../config/os/{generic => amigaos}/ctype_base.h     |  2 +-
 .../os/{generic => amigaos}/ctype_configure_char.cc |  2 +-
 .../config/os/{generic => amigaos}/ctype_inline.h   |  2 +-
 .../os/{generic => amigaos}/error_constants.h       |  2 +-
 .../config/os/{vxworks => amigaos}/os_defines.h     | 13 +++++++------
 libstdc++-v3/configure.host                         |  3 +++
 6 files changed, 14 insertions(+), 10 deletions(-)
 copy libstdc++-v3/config/os/{generic => amigaos}/ctype_base.h (97%)
 copy libstdc++-v3/config/os/{generic => amigaos}/ctype_configure_char.cc (97%)
 copy libstdc++-v3/config/os/{generic => amigaos}/ctype_inline.h (98%)
 copy libstdc++-v3/config/os/{generic => amigaos}/error_constants.h (98%)
 copy libstdc++-v3/config/os/{vxworks => amigaos}/os_defines.h (86%)

diff --git a/libstdc++-v3/config/os/generic/ctype_base.h b/libstdc++-v3/config/os/amigaos/ctype_base.h
similarity index 97%
copy from libstdc++-v3/config/os/generic/ctype_base.h
copy to libstdc++-v3/config/os/amigaos/ctype_base.h
index 9ac0d6be28dd9d7c58d7a6e803cd61222020979b..260b6142b6fc6a11a1e5f0bf7820b592d09f285e 100644
--- a/libstdc++-v3/config/os/generic/ctype_base.h
+++ b/libstdc++-v3/config/os/amigaos/ctype_base.h
@@ -1,9 +1,9 @@
 // Locale support -*- C++ -*-
 
-// Copyright (C) 1997-2016 Free Software Foundation, Inc.
+// Copyright (C) 1997-2015 Free Software Foundation, Inc.
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
 // terms of the GNU General Public License as published by the
 // Free Software Foundation; either version 3, or (at your option)
 // any later version.
diff --git a/libstdc++-v3/config/os/generic/ctype_configure_char.cc b/libstdc++-v3/config/os/amigaos/ctype_configure_char.cc
similarity index 97%
copy from libstdc++-v3/config/os/generic/ctype_configure_char.cc
copy to libstdc++-v3/config/os/amigaos/ctype_configure_char.cc
index af2012e53d9259444c97434323705aa811ba521e..e6f4895aee78da54bc6b1e01df00816206361c41 100644
--- a/libstdc++-v3/config/os/generic/ctype_configure_char.cc
+++ b/libstdc++-v3/config/os/amigaos/ctype_configure_char.cc
@@ -1,9 +1,9 @@
 // Locale support -*- C++ -*-
 
-// Copyright (C) 2011-2016 Free Software Foundation, Inc.
+// Copyright (C) 2011-2015 Free Software Foundation, Inc.
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
 // terms of the GNU General Public License as published by the
 // Free Software Foundation; either version 3, or (at your option)
 // any later version.
diff --git a/libstdc++-v3/config/os/generic/ctype_inline.h b/libstdc++-v3/config/os/amigaos/ctype_inline.h
similarity index 98%
copy from libstdc++-v3/config/os/generic/ctype_inline.h
copy to libstdc++-v3/config/os/amigaos/ctype_inline.h
index 3347d3324d2cd7de0e1badbf1855023762521d4c..cfa0146ae6b70623c2fe63b864ceef7bce0d5fe0 100644
--- a/libstdc++-v3/config/os/generic/ctype_inline.h
+++ b/libstdc++-v3/config/os/amigaos/ctype_inline.h
@@ -1,9 +1,9 @@
 // Locale support -*- C++ -*-
 
-// Copyright (C) 2000-2016 Free Software Foundation, Inc.
+// Copyright (C) 2000-2015 Free Software Foundation, Inc.
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
 // terms of the GNU General Public License as published by the
 // Free Software Foundation; either version 3, or (at your option)
 // any later version.
diff --git a/libstdc++-v3/config/os/generic/error_constants.h b/libstdc++-v3/config/os/amigaos/error_constants.h
similarity index 98%
copy from libstdc++-v3/config/os/generic/error_constants.h
copy to libstdc++-v3/config/os/amigaos/error_constants.h
index f5bb12c3b1446d4fb176ed0e641640580f73c7a5..74492cf61c875eac28cb8a86d5bfdb3b8807aa16 100644
--- a/libstdc++-v3/config/os/generic/error_constants.h
+++ b/libstdc++-v3/config/os/amigaos/error_constants.h
@@ -1,9 +1,9 @@
 // Specific definitions for generic platforms  -*- C++ -*-
 
-// Copyright (C) 2007-2016 Free Software Foundation, Inc.
+// Copyright (C) 2007-2015 Free Software Foundation, Inc.
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
 // terms of the GNU General Public License as published by the
 // Free Software Foundation; either version 3, or (at your option)
 // any later version.
diff --git a/libstdc++-v3/config/os/vxworks/os_defines.h b/libstdc++-v3/config/os/amigaos/os_defines.h
similarity index 86%
copy from libstdc++-v3/config/os/vxworks/os_defines.h
copy to libstdc++-v3/config/os/amigaos/os_defines.h
index 6bbce600cdabe71edad98bf90931c669f058c69a..346f063958cd7e80ebf97be4acee0bdf391cb811 100644
--- a/libstdc++-v3/config/os/vxworks/os_defines.h
+++ b/libstdc++-v3/config/os/amigaos/os_defines.h
@@ -1,9 +1,9 @@
-// Specific definitions for VxWorks  -*- C++ -*-
+// Specific definitions for AmigaOS -*- C++ -*-
 
-// Copyright (C) 2000-2016 Free Software Foundation, Inc.
+// Copyright (C) 2000-2015 Free Software Foundation, Inc.
 //
 // This file is part of the GNU ISO C++ Library.  This library is free
 // software; you can redistribute it and/or modify it under the
 // terms of the GNU General Public License as published by the
 // Free Software Foundation; either version 3, or (at your option)
 // any later version.
@@ -30,13 +30,14 @@
 #ifndef _GLIBCXX_OS_DEFINES
 #define _GLIBCXX_OS_DEFINES 1
 
 // System-specific #define, typedefs, corrections, etc, go here.  This
 // file will come before all others.
 
-//Keep vxWorks from defining min()/max() as macros
-#ifdef NOMINMAX
-#undef NOMINMAX
+// No ioctl() on AmigaOS
+#define _GLIBCXX_NO_IOCTL 1
+
+#ifdef __NEWLIB__
+#define _GLIBCXX_USE_C99_STDINT_TR1 1
 #endif
-#define NOMINMAX 1
 
 #endif
diff --git a/libstdc++-v3/configure.host b/libstdc++-v3/configure.host
index 304a7f5aff61d13ea4aba80e5b8261c192307579..72438701acdd434a0cb15c278f0dcd46756844e9 100644
--- a/libstdc++-v3/configure.host
+++ b/libstdc++-v3/configure.host
@@ -223,12 +223,15 @@ case "${host_os}" in
     atomicity_dir="os/aix"
     ;;
   aix*)
     os_include_dir="os/generic"
     atomicity_dir="cpu/generic"
     ;;
+  amigaos*)
+    os_include_dir="os/amigaos"
+    ;;
   bsd*)
     # Plain BSD attempts to share FreeBSD files.
     os_include_dir="os/bsd/freebsd"
     ;;
   cygwin*)
     os_include_dir="os/newlib"
-- 
2.34.1

